rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -2279,3 +2279,154 @@ rocalToDecibels(
     }
     return output;
 }
+
+RocalTensor ROCAL_API_CALL
+rocalResample(RocalContext p_context,
+              RocalTensor p_input,
+              RocalTensor p_output_resample_rate,
+              RocalTensorOutputType output_datatype,
+              bool is_output,
+              float sample_hint,
+              float quality) {
+    Tensor* output = nullptr;
+    if ((p_context == nullptr) || (p_input == nullptr) || (p_output_resample_rate == nullptr)) {
+        ERR("Invalid ROCAL context or invalid input tensor")
+        return output;
+    }
+    Tensor* resampled_output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    auto output_resample_rate = static_cast<Tensor*>(p_output_resample_rate);
+    try {
+        TensorInfo output_info = input->info();
+        RocalTensorDataType op_tensor_data_type = static_cast<RocalTensorDataType>(output_datatype);
+        if (op_tensor_data_type != RocalTensorDataType::FP32) {
+            WRN("Only FP32 dtype is supported for resample augmentation.")
+            op_tensor_data_type = RocalTensorDataType::FP32;
+        }
+        output_info.set_data_type(op_tensor_data_type);
+        if (sample_hint > 0) {
+            std::vector<size_t> max_dims = output_info.max_shape();
+            std::vector<size_t> dims = output_info.dims();
+            dims[1] = std::ceil(sample_hint);
+            dims[2] = max_dims[1];
```

Comment:
dims[2] is the same as max_dims[1] right?

Please check and we need not update that